### DISCUSSION TITLE
Potential for non-linear correlations between parameters?

### DISCUSSION THREAD

**Discussion by sgbaird** [sparks-baird/self-driving-lab-demo#56] [Demo extensions and design]
"Mixing" usually implies first-order, linear relationships. What about higher-order correlations between input parameters? Is there a way this could be introduced physically or artificially into the setup?

I need to think about this one more.

**Comment by sgbaird** [comment]
The more I think about it, the less it seems likely that this can be achieved physically in a low-cost setup. There's an option of mounting the LEDs to a [mini pan-tilt camera](https://www.adafruit.com/product/1967) (which introduces some very basic robotics), which would have some interesting effects. While this might introduce non-linearity, it could still probably be easily solved by gradient descent (i.e. single optimum).

**Comment by sgbaird** [comment]
Independently controlling the voltage and current of individual LEDs rather than using a prebuilt library that sets brightness might be an option. Of course, would require reasonable limits so the hardware doesn't get fried easily. More work. More custom.

**Comment by sgbaird** [comment]
LED current can be controlled within the AS7341 module https://github.com/adafruit/Adafruit_CircuitPython_AS7341/issues/1, but that's for the sensor, not the Blinkt! LED inputs. Perhaps Blinkt! also has an option to control LED current, though I'd be surprised if it had facilities for controlling the LED voltage.

**Comment by iterateccvoelker** [comment]
Hi @sgbaird, 

You could do a multi-objective optimisation where on the one hand you try to optimise the underlying physical function and on the other hand you try to match it with known designs that have certain properties. These properties could include remaining lifetime, CO2 efficiency, cost, etc. The goal could then be to find designs that have as similar characteristics as possible while being cheap, environmentally friendly, etc (inverse design)

Control parameters could be characteristic spectra (fingerprints), luminosity (achievable by higher power or reflectors), noise (flickering), etc.. The correlation of these quantities can be modelled using arbitrary functions to create complex correlations. 

Bottom line: you can model both the physical system and the design and optimise the two independent systems together. This could lead to interesting solutions where a slightly worse fingerprint match has a big advantage in terms of CO2 friendliness. The closer this is to a real engineering problem, the better.

**Comment by sgbaird** [comment]
I like the idea of having a stochastic, physically determined objective and a deterministic one like cost. This has been on my mind, especially with e.g. LED power.

> The closer this is to a real engineering problem, the better.

Definitely agreed! Thanks for weighing in on this.

**Comment by iterateccvoelker** [comment]
Right? 

We are currently building a demonstrator in which we simulate a disruption in the supply chain for alloys used in turbine blade printing. This is to show the improved resilience of a data-driven R&D lab. All we need to do is change the delivery times or costs of the precursors or processes and the deck of possible solutions is completely reshuffled. I will post a link as soon as it goes public.

**Comment by sgbaird** [comment]
> I will post a link as soon as it goes public.

Please do! Sounds very interesting

**Comment by sgbaird** [comment]
https://twitter.com/AlexBlokhuis/status/1583711930803376128

> SnCl2•2H2O dissolving in moisture picked up by deliquescent FeCl3•7H2O. On its own, the latter gives an orange drop. Reaction with SnCl2 makes this color disappear.

Maybe an interesting case-study for https://github.com/sparks-baird/self-driving-lab-demo/discussions/89
